### ISSUE TITLE
Error: No value present

### ISSUE THREAD

**Issue by estensen** [hyperledger/besu#4383]
Mainnet
v22.7.2
Checkpoint sync
Bonsai
Paired with Lighthouse

```
2022-09-12 12:51:11.628+02:00 | EthScheduler-Workers-2 | INFO  | PersistBlockTask | Imported #15,520,471 / 178 tx / 0 om / 13,734,857 (45.8%) gas / (0x513857dbcfa8ecdd18ec1614aa4205b0e956ec150e91e737e484348408fa5011) in 1.018s. Peers: 73
2022-09-12 12:51:33.339+02:00 | vert.x-eventloop-thread-3 | ERROR | ContextImpl | Unhandled exception
java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.get(Optional.java:143)
	at org.hyperledger.besu.ethereum.eth.transactions.sorter.BaseFeePendingTransactionsSorter.lambda$new$0(BaseFeePendingTransactionsSorter.java:76)
	at java.base/java.util.Comparator.lambda$comparing$77a9974f$1(Comparator.java:473)
	at java.base/java.util.Comparator.lambda$thenComparing$36697e65$1(Comparator.java:221)
	at java.base/java.util.Comparator.lambda$thenComparing$36697e65$1(Comparator.java:220)
	at java.base/java.util.Collections$ReverseComparator2.compare(Collections.java:5405)
	at java.base/java.util.TreeMap.getEntryUsingComparator(TreeMap.java:374)
	at java.base/java.util.TreeMap.getEntry(TreeMap.java:344)
	at java.base/java.util.TreeMap.remove(TreeMap.java:881)
	at java.base/java.util.TreeSet.remove(TreeSet.java:276)
	at org.hyperledger.besu.ethereum.eth.transactions.sorter.BaseFeePendingTransactionsSorter.doRemoveTransaction(BaseFeePendingTransactionsSorter.java:107)
	at org.hyperledger.besu.ethereum.eth.transactions.sorter.AbstractPendingTransactionsSorter.removeTransaction(AbstractPendingTransactionsSorter.java:170)
	at org.hyperledger.besu.ethereum.eth.transactions.sorter.AbstractPendingTransactionsSorter.lambda$evictOldTransactions$1(AbstractPendingTransactionsSorter.java:137)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3612)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.hyperledger.besu.ethereum.eth.transactions.sorter.AbstractPendingTransactionsSorter.evictOldTransactions(AbstractPendingTransactionsSorter.java:134)
	at org.hyperledger.besu.ethereum.eth.transactions.TransactionPoolEvictionService.lambda$start$0(TransactionPoolEvictionService.java:38)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:889)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:860)
	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:50)
	at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:274)
	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:22)
	at io.vertx.core.impl.AbstractContext.emit(AbstractContext.java:53)
	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:22)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.run(VertxImpl.java:883)
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:159)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.kqueue.KQueueEventLoop.run(KQueueEventLoop.java:293)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
2022-09-12 12:51:34.803+02:00 | EthScheduler-Workers-0 | INFO  | PersistBlockTask | Imported #15,520,472 / 317 tx / 0 om / 24,743,633 (82.5%) gas / (0x25ead51024d665c7928148cbd5f7a070b535d947a87230bedfe14cf710b02d4d) in 2.028s. Peers: 73
```

**Comment by macfarla** [comment]
Thanks for reporting this @estensen we will look into this - looks like your node keeps syncing from there though?

**Comment by estensen** [comment]
It does recover and keep syncing

**Comment by macfarla** [comment]
Think this should be fixed by https://github.com/hyperledger/besu/pull/4390

**Comment by macfarla** [comment]
Think this should be fixed by #4390
